lib/explorer/backend/series.ex

```diff
@@ -139,6 +139,16 @@ defmodule Explorer.Backend.Series do
   @callback window_max(s, window_size :: integer(), [window_option()]) :: s
   @callback window_mean(s, window_size :: integer(), [window_option()]) :: s
 
+  # Exponentially weighted windows
+
+  @type ewm_option() ::
+          {:alpha, float()}
+          | {:adjust, boolean()}
+          | {:min_periods, integer()}
+          | {:ignore_nulls, boolean()}
+
+  @callback ewm_mean(s, [ewm_option()]) :: s
```

Comment:
I liked the approach, and I think we could use more in our functions, instead of requiring each option to be an argument :+1:

Reply:
Sorry, I couldn't get what you suggest here.

Reply:
Oh, it's just a suggestion that we should do more of what you did here :)
I think I'm going to change some of our callbacks to this style. Specially the callbacks related to IO.

Reply:
There is a downside in this approach: if we add new options in the future, then existing implementations may forget to process said new options. Because of that, I am thinking we should perhaps go with the existing approach of passing each option as argument. Thoughts @philss?

Reply:
@josevalim you got me. Yeah, this downside is a problem. But on the other hand, it would give more flexibility and compatibility between versions. For now I think we could stick to the existing approach, and revisit this when we have another backend planned.

@sasikumar87 could you change this callback to use arguments instead of keyword args?

Reply:
@philss @josevalim Updated.
But `window_*` functions still use this kind of approach, should change them too?

Reply:
Yes, a PR that changes that will be welcome. Maybe instead of:

    do: Shared.apply_impl(series, :window_max, [window_size, window_opts_with_defaults(opts)])

We could do:

    do: Shared.apply_impl(series, :window_max, [window_size | window_args(opts)])

and `window_args` returns a list.